core/src/test/java/org/apache/brooklyn/core/mgmt/rebind/RebindConfigInheritanceTest.java

```diff
@@ -0,0 +1,145 @@
+        rebind();
+        rebindedApp = (Application) newManagementContext.lookup(entityId);
+        
+        // we'll also make assertions on the contents written out
+        RebindTestUtils.waitForPersisted(mgmt());
+        newMemento = Joiner.on("\n").join(Files.readLines(persistedEntityFile, Charsets.UTF_8));
+    }
+    
+    /** also see {@link RebindWithDeserializingClassRenamesTest} */
+    @Test
+    public void testPreBasicConfigInheritance_2016_07() throws Exception {
+        doReadConfigInheritance("prebasic-2016-07", "toruf2wxg4");
+        
+        ConfigKey<?> k = Iterables.getOnlyElement( rebindedApp.config().findKeys(ConfigPredicates.nameEqualTo("my.config.inheritanceMerged")) );
+        
+        Asserts.assertStringContains(origMemento, "<parentInheritance class=\"org.apache.brooklyn.config.ConfigInheritance$Merged\"/>");
+        Asserts.assertStringDoesNotContain(origMemento, BasicConfigInheritance.DEEP_MERGE.getClass().getName());
+
+        // should now convert it to BasicConfigInheritance.DEEP_MERGE
+        Asserts.assertStringDoesNotContain(newMemento, "ConfigInheritance$Merged");
+        Asserts.assertStringDoesNotContain(newMemento, "ConfigInheritance$Legacy$Merged");
+        Asserts.assertStringContains(newMemento, BasicConfigInheritance.DEEP_MERGE.getClass().getName());
+        
+        ConfigInheritance inh = k.getInheritanceByContext(InheritanceContext.RUNTIME_MANAGEMENT);
+        Assert.assertEquals(inh, BasicConfigInheritance.DEEP_MERGE);
+    }
+    
+    @Test
+    public void testBasicConfigInheritance_2016_10() throws Exception {
+        doReadConfigInheritance("basic-2016-10", "wj5s8u9h73");
+
+        checkNewAppNonInheritingKey1(rebindedApp);
+        
+        Asserts.assertStringContains(origMemento, "isReinherited");
+        Asserts.assertStringDoesNotContain(origMemento, "NEVER_INHERITED");
+        Asserts.assertStringDoesNotContain(origMemento, "NeverInherited");
+        
+        // should write it out as NeverInherited
+        Asserts.assertStringDoesNotContain(newMemento, "isReinherited");
+        Asserts.assertStringContains(newMemento, "NeverInherited");
+    }
+
+    @Test
+    public void testReadConfigInheritance_2016_11() throws Exception {
+        doReadConfigInheritance("basic-2016-11", "kmpez5fznt");
+        checkNewAppNonInheritingKey1(rebindedApp);
+        
+        String origMementoWithoutLicenseHeader = origMemento.substring(origMemento.indexOf("<entity>"));
+        Asserts.assertEquals(origMementoWithoutLicenseHeader, newMemento);
+    }
+    
+    @Test
+    public void testBasicConfigInheritanceProgrammatic() throws Exception {
+        origApp.config().set(key1, "1");
+        
+        RebindOptions opts = RebindOptions.create();
+        opts.keepSameInstance = true;
+        
+        rebind(opts);
+        
+        String entityFile = Streams.readFully(new FileReader(new File(opts.mementoDir, "entities/"+origApp.getApplicationId())));
```

Comment:
d'oh.  thought it was awfully hard to find that!  agree, will remove keepSameInstance.